Marlin/Marlin_main.cpp

```diff
@@ -3940,6 +3940,10 @@ inline void gcode_M109() {
   // Exit if the temperature is above target and not waiting for cooling
   if (no_wait_for_cooling && !isHeatingHotend(target_extruder)) return;
 
+  // Prevents a wait-forever situation if R is misused i.e. M109 R0
+  // Try to calculate a ballpark safe margin by halving EXTRUDE_MINTEMP
+  if (degTargetHotend(target_extruder) < (EXTRUDE_MINTEMP/2)) return;
```

Comment:
Because the macro `EXTRUDE_MINTEMP` could be set by a user to something like "5 + 5" the line should actually look like this to avoid an expansion like "`(5+5/2)`":
```cpp
if (degTargetHotend(target_extruder) < (EXTRUDE_MINTEMP) / 2) return;
```